File: octal_to_decimal/CPP/src/oct_to_dectest.cpp
#include "../include/oct_to_dec.hpp"
#include "../include/oct_to_dectest.hpp"
#include <iostream>
#include <vector>


bool OctToDecimalTester::testOctToDecimal(std::string octal, int expectedDecimal) {
    OctToDecimal converter;
    int result = converter.octToDecimal(octal);
    return result == expectedDecimal;
}

// This file contains the test logic without the main method.

File: octal_to_decimal/JAVA/src/oct_to_dectest.java
public class oct_to_dectest {
    boolean test(String input, int expectedOutput) {
        if (input == null) {
            throw new IllegalArgumentException("Input octal string cannot be null");
        }
        
        oct_to_dec octToDec = new oct_to_dec();
        int result = octToDec.convertToDecimal(input);
        return result == expectedOutput;
    }
}